### PR TITLE
refactor the options and start function of factory

### PR DESCRIPTION
**Pull request by CoderZhi**
Move the redundant code to Start function

### CODE REVIEW

File: state/factory/factory.go [hunk 1]
```diff
@@ -173,8 +143,19 @@ func NewFactory(cfg config.Config, opts ...Option) (Factory, error) {
 func (sf *factory) Start(ctx context.Context) error {
 	sf.mutex.Lock()
 	defer sf.mutex.Unlock()
+	var err error
+	if err = sf.lifecycle.OnStart(ctx); err != nil {
```
Comment (zjshen14) [inline state/factory/factory.go]: Do you need to remove L126-L128? Otherwise, sf.accountTrie will be started as well by this call?

Comment (CoderZhi) [reply state/factory/factory.go]: addressed

File: state/factory/factory.go [hunk 2]
```diff
@@ -143,19 +143,8 @@ func NewFactory(cfg config.Config, opts ...Option) (Factory, error) {
...
-	// get state trie root
-	if sf.rootHash, err = sf.getRoot(trie.AccountKVNameSpace, AccountTrieRootKey); err != nil {
-		return errors.Wrap(err, "failed to get accountTrie's root hash from underlying DB")
-	}
-	if sf.accountTrie, err = trie.NewTrie(sf.dao, trie.AccountKVNameSpace, sf.rootHash); err != nil {
-		return errors.Wrap(err, "failed to generate accountTrie from config")
-	}
 
-	return sf.accountTrie.Start(ctx)
+	return sf.lifecycle.OnStart(ctx)
```
Comment (zjshen14) [inline state/factory/factory.go]: So the order doesn't matter?

Comment (CoderZhi) [reply state/factory/factory.go]: The order doesn't matter as long as we handle the dependency properly. In our previous implementation, the root hash was stored in both sf and trie, which results in the messy of dependency.

**Pull request merged**